web/src/lib/roadmap/roadmapBuilder.ts

```diff
@@ -128,37 +131,32 @@ const findTaskInRoadmapByFilename = (
   roadmapBuilder: RoadmapBuilder,
   taskFilename: string
 ): TaskBuilder | undefined => {
-  for (const step of roadmapBuilder.steps) {
-    const found = step.tasks.find((task) => task.filename === taskFilename);
-    if (found) return found;
-  }
+  const found = roadmapBuilder.tasks.find((task) => task.filename === taskFilename);
+  if (found) return found;
 };
 
 const convertToRoadmap = async (roadmapBuilder: RoadmapBuilder): Promise<Roadmap> => {
   const roadmap = {
-    steps: await Promise.all(
-      roadmapBuilder.steps.map(async (step) => ({
-        ...step,
-        tasks: await Promise.all(
-          step.tasks.sort(orderByWeight).map((task) => fetchTaskByFilename(task.filename))
-        ),
-      }))
+    steps: await Promise.all(roadmapBuilder.steps),
+    tasks: await Promise.all(
+      roadmapBuilder.tasks.sort(orderByWeight).map((task) => fetchTaskByFilename(task.filename))
     ),
   };
 
-  const allFilenames = roadmap.steps.reduce(
-    (acc: string[], currStep: Step) => [...acc, ...currStep.tasks.map((task) => task.filename)],
-    []
+  roadmap.tasks.map(
+    (task: Task) =>
+      (task.stepNumber = (
+        findTaskInRoadmapByFilename(roadmapBuilder, task.filename) as TaskBuilder
+      ).step_number)
   );
 
+  const allFilenames = new Set(roadmap.tasks.map((task) => task.filename));
```

Comment:
why a set? they should already be all unique right?

Reply:
es-lint kept complaining and suggested the refactor to a `Set` and using `has`.